src/rnpkeys/rnpkeys.cpp

```diff
@@ -53,6 +53,7 @@ const char *usage =
   "    --expiration         Set key and subkey expiration time.\n"
   "    --cipher             Set cipher used to encrypt a secret key.\n"
   "    --hash               Set hash which is used for key derivation.\n"
+  "    --force-weak         Force usage of a weak hash algorithm.\n"
```

Comment:
Is `allow-weak` a better name than `force-weak`?

Reply:
Agree, `allow` should look better here, as `force` could be confusing to user (making think that only weak hashes should be used). Also probably it should be extended to `--allow-weak-hashes` since we would also need to have `--allow-weak-ciphers` at some point.

Reply:
Ping @antonsviridenko for an update. Thanks!

Reply:
> Also probably it should be extended to --allow-weak-hashes since we would also need to have --allow-weak-ciphers at some point.

Single option can cover both. Is there any reason to handle weak hashes and ciphers separately?

Reply:
> Single option can cover both. Is there any reason to handle weak hashes and ciphers separately?

It is always better to give user flexibility. This case could lead to unwanted or unexpected consequence (for instance, user would like to use SHA1 but accidentally would enable IDEA and use it for encryption).

Reply:
@ni4 maybe `--allow-weak-hash` is better?  `--allow-weak-hashes` is too long and I'll have to shift the whole right  column in the usage text otherwise :)
```
 "    --hash             Specify hash algorithm, used during signing.\n"
 "    --allow-weak-hashesAllow usage of a weak hash algorithm.\n"


```

Reply:
@antonsviridenko sure, --allow-weak-hash would work as well.

Reply:
Agree!